File: MATTHEW/leetcode/AIcoding/5.Qwen-3/javascript_solutions/826-most-profit-assigning-work.js
function maxProfitAssign(workers, difficulty, profit, worker) {
    const jobs = difficulty.map((d, i) => [d, profit[i]]);
    jobs.sort((a, b) => a[0] - b[0]);

    for (let i = 1; i < jobs.length; i++) {
        jobs[i][1] = Math.max(jobs[i][1], jobs[i - 1][1]);
    }

    let res = 0;
    let j = jobs.length - 1;

    for (let i = workers.length - 1; i >= 0; i--) {
        while (j >= 0 && workers[i] >= jobs[j][0]) {
            res = Math.max(res, jobs[j][1]);
            j--;
        }
    }

    return res;
}

File: leetcode/AIcoding/5.Qwen-3/cpp_solutions/826-most-profit-assigning-work.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int maxProfitAssignment(vector<int>& difficulty, vector<int>& profit, vector<int>& worker) {
        vector<pair<int, int>> jobs;
        for (int i = 0; i < difficulty.size(); ++i) {
            jobs.push_back({difficulty[i], profit[i]});
        }
        sort(jobs.begin(), jobs.end());
        sort(worker.begin(), worker.end());

        int res = 0;
        int j = 0;
        int maxProfit = 0;

        for (int w : worker) {
            while (j < jobs.size() && jobs[j].first <= w) {
                maxProfit = max(maxProfit, jobs[j].second);
                ++j;
            }
            res += maxProfit;
        }

        return res;
    }
};